frontend/appflowy_flutter/lib/user/presentation/screens/sign_in_screen/desktop_sign_in_screen.dart

```diff
@@ -1,62 +1,79 @@
 
 class DesktopSignInScreen extends StatelessWidget {
-  const DesktopSignInScreen({super.key, required this.isLoading});
-
-  final bool isLoading;
+  const DesktopSignInScreen({
+    super.key,
+  });
 
   @override
   Widget build(BuildContext context) {
     const indicatorMinHeight = 4.0;
-    return Scaffold(
-      appBar: const PreferredSize(
-        preferredSize: Size(double.infinity, 60),
-        child: MoveWindowDetector(),
-      ),
-      body: Center(
-        child: AuthFormContainer(
-          children: [
-            FlowyLogoTitle(
-              title: LocaleKeys.welcomeText.tr(),
-              logoSize: const Size(60, 60),
-            ),
-            const VSpace(30),
+    return BlocBuilder<SignInBloc, SignInState>(
+      builder: (context, state) {
+        return Scaffold(
+          appBar: const PreferredSize(
+            preferredSize: Size(double.infinity, 60),
+            child: MoveWindowDetector(),
+          ),
+          body: Center(
+            child: AuthFormContainer(
+              children: [
+                FlowyLogoTitle(
+                  title: LocaleKeys.welcomeText.tr(),
+                  logoSize: const Size(60, 60),
+                ),
+                const VSpace(30),
 
-            // const SignInAnonymousButton(),
-            const SignInWithMagicLinkButtons(),
+                // const SignInAnonymousButton(),
+                const SignInWithMagicLinkButtons(),
 
-            // third-party sign in.
-            const VSpace(20),
+                // third-party sign in.
+                const VSpace(20),
 
-            if (isAuthEnabled) ...[
-              const _OrDivider(),
-              const VSpace(10),
-              const ThirdPartySignInButtons(),
-            ],
-            const VSpace(20),
+                if (isAuthEnabled) ...[
+                  const _OrDivider(),
+                  const VSpace(10),
+                  const ThirdPartySignInButtons(),
+                ],
+                const VSpace(20),
 
-            // anonymous sign in
-            const SignInAnonymousButtonV2(),
+                // anonymous sign in
+                const SignInAnonymousButtonV2(),
+                const VSpace(10),
 
-            // loading status
-            const VSpace(indicatorMinHeight),
-            isLoading
-                ? const LinearProgressIndicator(
-                    minHeight: indicatorMinHeight,
-                  )
-                : const VSpace(indicatorMinHeight),
-            const VSpace(20),
-          ],
-        ),
-      ),
+                SwitchSignInSignUpButton(
+                  onTap: () {
+                    final type = state.loginType == LoginType.signIn
+                        ? LoginType.signUp
+                        : LoginType.signIn;
+                    context.read<SignInBloc>().add(
+                          SignInEvent.switchLoginType(type),
+                        );
+                  },
```

Comment:
```suggestion
                  onTap: () => context.read<SignInBloc>().add(
                          const SignInEvent.toggleLoginType(),
                        );
```
_personal preference_